packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -274,7 +275,7 @@ export default class Dotnet extends Target {
     assm: spec.Assembly,
   ): { [language: string]: PackageInfo } {
     const packageId = assm.targets!.dotnet!.packageId;
-    const version = assm.version;
+    const version = toReleaseVersion(assm.version, 'dotnet');
```

Comment:
Something is off here in terms of code cleanliness. 

Ideally, besides the jsii `Assembly` context, there's also a pacmak specific context that gets passed around, which converts `assembly.version` into the release version identifier. So the conversion of one version to the other happens only once. Something like -

```ts
interface PacMakContext {
  public readonly assm: spec.Assembly;
  // ...
  public readonly releaseVersion;
}
```

This would "home" the translation in the correct place.

Do you see value in something like that?

Reply:
Could be. I mean this is also why long term I'm interested in having code-gen happen within `jsii` directly...

Reply:
Also - the problem with your suggestion is that it would need to carry a context... for *each* dependency, too. Because we also need to perform the prerelease mapping for dependency ranges (otherwise we'll have resolution issues)

Reply:
I'll let you decide how useful or not it is. I've barely worked in this side of the code base.
Just feels like something we could centralize `toReleaseVersion()` and make it available via a context.